packages/block-library/src/quote/deprecated.js

```diff
@@ -112,6 +106,47 @@ const deprecated = [
 			);
 		},
 	},
+	{
+		attributes: {
+			...blockAttributes,
+			value: {
+				type: 'string',
+				source: 'html',
+				selector: 'blockquote',
+				multiline: 'p',
+				default: '',
+			},
+		},
+		migrate( attributes ) {
+			// The old value attribute for quotes can contain:
+			// - a single paragraph: "<p>single paragraph</p>"
+			// - multiple paragraphs: "<p>first paragraph</p><p>second paragraph</p>"
+			const innerBlocks = parseWithAttributeSchema( attributes.value, {
+				type: 'array',
+				source: 'query',
+				selector: 'p',
+				query: {
+					value: {
+						type: 'string',
+						source: 'text',
+					},
+				},
+			} ).map( ( { value } ) =>
+				createBlock( 'core/paragraph', { content: value } )
+			);
+			return [ omit( attributes, [ 'value' ] ), innerBlocks ];
+		},
+		save( { attributes: { value, citation } } ) {
+			return (
+				<blockquote>
+					<RichText.Content multiline value={ value } />
+					{ ! RichText.isEmpty( citation ) && (
+						<RichText.Content tagName="cite" value={ citation } />
+					) }
+				</blockquote>
```

Comment:
Now is probably a good time to point out that the latest spec for the `<blockquote>` element makes this markup incorrect:

https://html.spec.whatwg.org/multipage/grouping-content.html#the-blockquote-element

The older W3C HTML 5.3 spec is a bit more lenient, but both it and the WHATWG spec include something like this as an example of how the blockquote "should" be given a citation:

```html
<figure>
 <blockquote>
  Blah blah blah
 </blockquote>
 <figcaption>Zebulan Stanphill, in "<cite>Book Title</cite>"</figcaption>
</figure>
```

The WHATWG spec states that since a citation is not part of the quote itself, it should therefore not be in the `<blockquote>`.

Right now is our best opportunity to make a markup change like this, since the markup is changing anyway because of this refactor. If the use of `figure` seems like too much, then I think we should at least switching to something like this, which is valid in the old W3C spec (but not the WHATWG spec):

```html
<blockquote>
Blah blah blah I like <cite>Book Name</cite>.
<footer><cite>Actual quote citation</cite></footer>
</blockquote>
```

This would at least allow the blockquotes to contain other `<cite>` elements within the quote itself (which are completely valid, since `<cite>`s are not just used for blockquote citations).

But personally, I prefer the latest version in the WHATWG spec. It seems like the most logical and flexible markup to me.

Reply:
I like the figure. Adds some consistency in our markup across blocks.

Reply:
If a quote doesn't have a source, should it still use a figure? Seems a bit excessive.

Reply:
We do the same for tables and media, but maybe we can't omit the figure if it's not present.

Reply:
by my reading of the spec I don't see any reason to add the `<figure>` if we don't have a citation. 

that is, if just containing quote material we get `<blockquote><!-- inner blocks --></blockquote>` but if we have a citation then `<figure><blockquote><!-- inner blocks --></blockquote><figcaption><!-- RichText citation --></figure>`

Reply:
now I see this was already discussed below. sorry.